internal/commands/request/request_internal.go

```diff
@@ -0,0 +1,294 @@
+		}
+	}
+
+	output.Print(output.Opts{
+		Message: "PingOne Access token does not exist or is expired, requesting a new token...",
+		Result:  output.ENUM_RESULT_NOACTION_WARN,
+	})
+
+	// If no valid access token is available, login and get a new one
+	return pingoneAuth()
+}
+
+func pingoneAuth() (accessToken string, err error) {
+	topLevelDomain, err := getTopLevelDomain()
+	if err != nil {
+		return "", err
+	}
+
+	workerEnvId, err := profiles.GetOptionValue(options.PingoneAuthenticationWorkerEnvironmentIDOption)
+	if err != nil {
+		return "", err
+	}
+
+	if workerEnvId == "" {
+		return "", fmt.Errorf("PingOne worker environment ID is required")
+	}
+
+	authURL := fmt.Sprintf("https://auth.pingone.%s/%s/as/token", topLevelDomain, workerEnvId)
+
+	clientId, err := profiles.GetOptionValue(options.PingoneAuthenticationWorkerClientIDOption)
+	if err != nil {
+		return "", err
+	}
+	clientSecret, err := profiles.GetOptionValue(options.PingoneAuthenticationWorkerClientSecretOption)
+	if err != nil {
+		return "", err
+	}
+
+	if clientId == "" || clientSecret == "" {
+		return "", fmt.Errorf("PingOne client ID and secret are required")
+	}
+
+	basicAuthBase64 := base64.StdEncoding.EncodeToString([]byte(clientId + ":" + clientSecret))
+
+	payload := strings.NewReader("grant_type=client_credentials")
+
+	client := &http.Client{}
+	req, err := http.NewRequest(customtypes.ENUM_HTTP_METHOD_POST, authURL, payload)
+	if err != nil {
+		return "", err
+	}
+
+	req.Header.Add("Authorization", fmt.Sprintf("Basic %s", basicAuthBase64))
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+
+	res, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+
+	defer res.Body.Close()
+	responseBodyBytes, err := io.ReadAll(res.Body)
+	if err != nil {
+		return "", err
+	}
+
+	if res.StatusCode < 200 || res.StatusCode >= 300 {
+		return "", fmt.Errorf("failed to authenticate with PingOne: Response Status %s: Response Body %s", res.Status, string(responseBodyBytes))
+	}
+
+	pingoneAuthResponse := new(PingoneAuthResponse)
+	err = json.Unmarshal(responseBodyBytes, pingoneAuthResponse)
+	if err != nil {
+		return "", err
+	}
+
+	// Store access token and expiry
+	profileViper := profiles.GetMainConfig().ActiveProfile().ViperInstance()
+	profileViper.Set(options.RequestAccessTokenOption.ViperKey, pingoneAuthResponse.AccessToken)
+
+	currentTime := time.Now().Unix()
+	tokenExpiry := currentTime + pingoneAuthResponse.ExpiresIn
+	profileViper.Set(options.RequestAccessTokenExpiryOption.ViperKey, tokenExpiry)
+
+	err = profiles.GetMainConfig().SaveProfile(profiles.GetMainConfig().ActiveProfile().Name(), profileViper)
+	if err != nil {
+		return "", err
+	}
+
+	return pingoneAuthResponse.AccessToken, nil
+}
+
+func getData() (data string, err error) {
+	data, err = profiles.GetOptionValue(options.RequestDataOption)
+	if err != nil {
+		return "", err
+	}
+
+	if data == "" {
+		return "", nil
+	}
+
+	// if data string first character is '@', read from file
```

Comment:
Is this a typical convention?

Reply:
This is the format specified by the acceptance criteria in PDI-1454. Unsure about the wider convention.